contrib/hit/parse.cc

```diff
@@ -188,12 +188,13 @@ Node::remove()
   delete this;
 }
 
-std::string
+const std::string &
 Node::name()
 {
   if (_toks.size() > 0)
-    return *_toks[0].name;
-  return "";
+    return _toks[0].name;
+  static std::string unknown_name = "[unknown]";
```

Comment:
Using a static to avoid returning a reference to a temporary local. I think we should return something other than an empty sting here or error message could end up looking weird (unless we just want to rely on file names always being enclosed in quotes in the message - in that case the user would see an empty pair of quotes which would indicate the absence of a name)

Reply:
A good improvement.